src/tty_interface.c

```diff
@@ -118,13 +118,18 @@ static void action_emit(tty_interface_t *state) {
 	/* ttyout should be flushed before outputting on stdout */
 	tty_close(state->tty);
 
-	const char *selection = choices_get(state->choices, state->choices->selection);
-	if (selection) {
-		/* output the selected result */
-		printf("%s\n", selection);
-	} else {
-		/* No match, output the query instead */
-		printf("%s\n", state->search);
+	unsigned int all = state->options->output_all;
+	size_t start     = all ? 0 : state->choices->selection;
+	size_t end       = all ? choices_available(state->choices) : start + 1;
```

Comment:
Plus having to select code and run the macro is also too much hassle, for it to be worth it (ignoring the source control issues) it would have to automatically detect consecutive assignments and update them all without needing any user selection.